e2e/testcases/ssh_known_hosts_test.go

```diff
@@ -0,0 +1,204 @@
+	var err error
+	rootSecret := &corev1.Secret{}
+	nt.T.Cleanup(func() {
+		nt.MustMergePatch(rootSecret, secretDataDeletePatch(controllers.KnownHostsKey))
+		nomostest.DeletePodByLabel(nt, "app", reconcilermanager.ManagerName, true)
+	})
+
+	// get known host key value
+	knownHostValue, err := nomostest.GetKnownHosts(nt)
+	if err != nil {
+		nt.T.Fatalf("error: %s", err)
+	}
+
+	// validate secret and deployment initial stage
+	err = nt.Validate(nomostest.RootAuthSecretName, configsync.ControllerNamespace, rootSecret)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		core.RootReconcilerPrefix, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "false"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// apply known host key and validate
+	nt.MustMergePatch(rootSecret, secretDataPatch(controllers.KnownHostsKey, knownHostValue))
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		core.RootReconcilerPrefix, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "true"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// try syncing resource and validate
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/accounting-namespace.yaml", yamlDir), "acme/namespaces/accounting/namespace.yaml"))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add accounting namespace"))
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Validate("accounting", "", &corev1.Namespace{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/accounting/namespace.yaml"))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove accounting namespace"))
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.ValidateNotFound("accounting", "", &corev1.Namespace{}); err != nil {
+		nt.T.Fatalf("error: %s", err)
+	}
+
+	// validate root sync error using invalid known host value
+	knownHostValue = "invalid value"
+	nt.MustMergePatch(rootSecret, secretDataPatch(controllers.KnownHostsKey, knownHostValue))
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		core.RootReconcilerPrefix, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "true"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	err = nt.Watcher.WatchObject(kinds.RootSyncV1Beta1(), configsync.RootSyncName, configsync.ControllerNamespace, []testpredicates.Predicate{
+		testpredicates.RootSyncHasSourceError(status.SourceErrorCode, "No ED25519 host key is known"),
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestRepoReconcilerSSHKnownHost(t *testing.T) {
```

Comment:
nit: we typically refer to RepoSync reconcilers as "namespace reconcilers" rather than "repo reconcilers"

Perhaps rename to `TestNsReconcilerSSHKnownHost`, although I think I prefer `TestRepoSyncSSHKnownHost`. If you opt for the latter, also rename the other test to `TestRootSyncSSHKnownHost` for consistency

Reply:
renamed to `TestRootSyncSSHKnownHost` and `TestRepoSyncSSHKnownHost`